provisioning/security/x509-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderX509Cert.java

```diff
@@ -8,73 +8,44 @@
 package com.microsoft.azure.sdk.iot.provisioning.security.hsm;
 
 import com.microsoft.azure.sdk.iot.provisioning.security.SecurityProviderX509;
-import com.microsoft.azure.sdk.iot.provisioning.security.exceptions.SecurityProviderException;
-import org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
-import org.bouncycastle.jce.provider.BouncyCastleProvider;
-import org.bouncycastle.openssl.PEMKeyPair;
-import org.bouncycastle.openssl.PEMParser;
-import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
-import org.bouncycastle.util.io.pem.PemObject;
-import org.bouncycastle.util.io.pem.PemReader;
 
-import java.io.ByteArrayInputStream;
-import java.io.IOException;
-import java.io.StringReader;
 import java.security.Key;
-import java.security.Security;
-import java.security.cert.CertificateException;
-import java.security.cert.CertificateFactory;
 import java.security.cert.X509Certificate;
 import java.util.Collection;
-import java.util.LinkedList;
 
 public class SecurityProviderX509Cert extends SecurityProviderX509
 {
     private static final String CN = "CN=";
     private static final String COMMA = ",";
     private static final String EQUALS = "=";
+
     private final String commonNameLeaf;
     private final X509Certificate leafCertificatePublic;
     private final Key leafPrivateKey;
     private final Collection<X509Certificate> signerCertificates;
 
-    private final String leafCertificatePublicPem;
-    private final Collection<String> signerCertificatesPem;
     /**
      * Constructor to build the DICE certs from the simulator
      */
-    public SecurityProviderX509Cert(String leafPublicPem, String leafPrivateKey, Collection<String> signerCertificates) throws SecurityProviderException
```

Comment:
This API needed to take the security primitives rather than strings because there is no way for us to parse these strings reliably without further details about what kinds of certificates/keys they are (RSA vs ECC, for instance)